roles/common/defaults/main.yml

```
@@ -13,7 +13,12 @@ ats_home: "{{ binaries_folder }}/transform-service"
 # Using default() filter for hostvars 1st level key requires ansible 2.10+
 nginx_host: "{% if groups.nginx | length == 0 %}127.0.0.1{% else %}{{ hostvars[groups.nginx[0]].ansible_host | default('127.0.0.1') }}{% endif %}"
 repo_host: "{% if groups.repository | length == 0 %}127.0.0.1{% else %}{{ hostvars[groups.repository[0]].ansible_default_ipv4.address | default('127.0.0.1') }}{% endif %}"
-activemq_host: "{% if groups.activemq | length == 0 %}127.0.0.1{% else %}{{ hostvars[groups.activemq[0]].ansible_default_ipv4.address | default('127.0.0.1') }}{% endif %}"
+activemq_host: >-
+  {% if 'external_activemq' in groups %}
```

```suggestion
  {% if groups.external_activemq %}
```
Otherwise if `external_active` is present but empty that'll break.

---

mine it's more defensive, when using attribute access you need to explicitly provide a `| default` or it will abort with `no such attribute` when the group is empty